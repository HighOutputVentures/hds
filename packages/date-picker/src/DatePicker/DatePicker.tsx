import { chakra } from '@chakra-ui/react';
import { addMonths, format, isEqual, subMonths } from 'date-fns';
import * as React from 'react';
import { v4 as uuid } from 'uuid';
import { DAYS } from '../constants';
import { useStyles } from '../hooks';
import { getCalendar, noop } from '../utils';
import { DatePickerControl } from './DatePickerControl';

export type DatePickerProps = {
  value?: Date;
  onChange?(selected: Date): void;
};

export function DatePicker({ value, onChange = noop }: DatePickerProps) {
  const [baseDate, setBaseDate] = React.useState(value ?? new Date());

  const calendar = React.useMemo(() => getCalendar(baseDate), [baseDate]);
  const styles = useStyles();

  return (
    <chakra.div sx={styles.calendar({ hasBorder: true })}>
      <DatePickerControl
        value={baseDate}
        onNext={() => setBaseDate((d) => addMonths(d, 1))}
        onPrev={() => setBaseDate((d) => subMonths(d, 1))}
        __nextButtonTestId="hds.datepicker.controls.next-month"
        __prevButtonTestId="hds.datepicker.controls.prev-month"
        __selectedMonthTestId="hds.datepicker.selected-month"
      />

      <chakra.table sx={styles.calendarMain()} data-testid="hds.datepicker.calendar">
        <chakra.thead>
          <chakra.tr>
            {DAYS.map((d) => (
              <chakra.th
                key={uuid()}
                sx={styles.calendarWeek()}
                data-testid="hds.datepicker.calendar.weekday"
              >
                {d}
              </chakra.th>
            ))}
          </chakra.tr>
        </chakra.thead>

        <chakra.tbody>
          {calendar.map((arr) => (
            <chakra.tr key={uuid()}>
              {arr.map(({ isToday, isPlaceholder, ...obj }) => {
                const formatted = format(obj.value, 'yyyy-MM-dd');
                const isSelected = !!value && isEqual(obj.value, value);

                return (
                  <chakra.td key={uuid()}>
                    <chakra.button
                      tabIndex={-1}
                      onClick={() => {
                        onChange(obj.value);
                        setBaseDate(obj.value);
                      }}
                      sx={styles.calendarDate({
                        isToday,
                        isSelected,
                        isPlaceholder,
                      })}
                      data-testid={`hds.datepicker.calendar.date.${formatted}`}
                    >
                      {obj.value.getDate()}
                    </chakra.button>
                  </chakra.td>
                );
              })}
            </chakra.tr>
          ))}
        </chakra.tbody>
      </chakra.table>
    </chakra.div>
  );
}
