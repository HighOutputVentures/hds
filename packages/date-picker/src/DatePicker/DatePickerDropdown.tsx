export type DatePickerDropdownProps = {};

export function DatePickerDropdown() {
  return null;
}
