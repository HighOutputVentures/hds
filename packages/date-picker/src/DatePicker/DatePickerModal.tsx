export type DatePickerModalProps = {};

export function DatePickerModal() {
  return null;
}
