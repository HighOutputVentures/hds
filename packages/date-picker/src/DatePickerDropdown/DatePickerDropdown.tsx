export type DatePickerDropdownProps = {};

/**
 *
 * NOT IMPLEMENTED
 *
 */
export function DatePickerDropdown() {
  return null;
}
