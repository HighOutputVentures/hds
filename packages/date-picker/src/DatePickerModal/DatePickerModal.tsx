export type DatePickerModalProps = {};

/**
 *
 * NOT IMPLEMENTED
 *
 */
export function DatePickerModal() {
  return null;
}
