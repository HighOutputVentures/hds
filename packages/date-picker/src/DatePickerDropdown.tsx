import { CalendarIcon } from '@chakra-ui/icons';
import {
  Button,
  Menu,
  MenuButton,
  MenuList,
  Text,
  useDisclosure,
} from '@chakra-ui/react';
import moment from 'moment';
import React from 'react';

import { dateInfoProps, DatePicker } from './DatePicker';
export interface DatePickerDropdownProps {
  type?: 'pre-set ranges' | 'dual dates' | 'single date';
  dateEvents?: Array<{ date: Date; title: string; id: string }>;
  userId?: string;

  onClose: () => void;
  isOpen: boolean;
  onApplyDate?: (props: dateInfoProps['info']) => void;
  timezone: string;
}
const DatePickerDropdown = (props: DatePickerDropdownProps) => {
  const { onApplyDate, dateEvents, timezone, type, userId } = props;
  const { isOpen, onClose, onOpen } = useDisclosure();
  const [selectedDate, setSelectedDate] = React.useState<dateInfoProps['info']>(
    []
  );

  return (
    <Menu isOpen={isOpen} onClose={onClose}>
      <MenuButton
        as={Button}
        onClick={onOpen}
        leftIcon={<CalendarIcon />}
        bg={'white'}
        color="black"
        border="1px solid #D6D6D6"
        boxShadow={'0px 1px 2px rgba(16, 24, 40, 0.05)'}
        borderRadius={'8px'}
        _hover={{ bg: 'none' }}
      >
        {selectedDate.length === 0 && 'Select Date'}
        {selectedDate.length >= 1 && (
          <Text as="span">
            {moment(
              `${selectedDate[0]?.year}-${selectedDate[0]?.month}-${selectedDate[0]?.day}`
            ).format('ll')}
          </Text>
        )}
        {selectedDate.length === 2 && <Text as="span"> {' â€“ '}</Text>}
        {selectedDate.length === 2 && (
          <Text as="span">
            {moment(
              `${selectedDate[1]?.year}-${selectedDate[1]?.month}-${selectedDate[1]?.day}`
            ).format('ll')}
          </Text>
        )}
      </MenuButton>
      <MenuList
        bg="none"
        w={'100%'}
        height={'100%'}
        border="none"
        data-testid={'hds.dropdown-datepicker'}
      >
        <DatePicker
          dateEvents={dateEvents}
          type={type}
          isOpen={true}
          onClose={onClose}
          userId={userId}
          timezone={timezone}
          onApplyDate={(d) => {
            setSelectedDate(d);
            onApplyDate;
          }}
        />
      </MenuList>
    </Menu>
  );
};

export default DatePickerDropdown;
