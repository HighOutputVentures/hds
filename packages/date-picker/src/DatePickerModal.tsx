import { CalendarIcon } from '@chakra-ui/icons';
import {
  Box,
  Button,
  Modal,
  ModalBody,
  ModalContent,
  ModalOverlay,
  Text,
  useDisclosure,
} from '@chakra-ui/react';
import moment from 'moment';
import React, { Fragment, ReactNode, useMemo } from 'react';
import { dateInfoProps, DatePicker } from './DatePicker';

type RenderChildren = (context: {
  value: dateInfoProps['info'];
  isOpen: boolean;
  onOpen(): void;
  onClose(): void;
  onToggle(): void;
}) => ReactNode;

export interface DatePickerModalProps {
  type?: 'pre-set ranges' | 'dual dates' | 'single date';
  userId?: string;
  dateEvents?: Array<{ date: Date; title: string; id: string }>;
  onApplyDate?: (props: dateInfoProps['info']) => void;
  timezone?: string;
  children?: RenderChildren;
}

const DatePickerModal = (props: DatePickerModalProps) => {
  const {
    type,
    userId,
    dateEvents,
    onApplyDate,
    timezone = 'Asia/Manila',
    children,
  } = props;

  const { isOpen, onClose, onOpen, onToggle } = useDisclosure();

  const [selectedDate, setSelectedDate] = React.useState<dateInfoProps['info']>(
    []
  );

  const value = useMemo(() => selectedDate, [selectedDate]);

  return (
    <Box data-testid={'hds.modal-datepicker'}>
      {!!children && (
        <Fragment>
          {children({
            value,
            isOpen,
            onOpen,
            onClose,
            onToggle,
          })}
        </Fragment>
      )}

      {!children && (
        <Button
          onClick={onOpen}
          leftIcon={<CalendarIcon />}
          bg={'white'}
          color="black"
          border="1px solid #D6D6D6"
          boxShadow={'0px 1px 2px rgba(16, 24, 40, 0.05)'}
          borderRadius={'8px'}
          _hover={{
            bg: 'none',
          }}
        >
          {selectedDate.length === 0 && 'Select Date'}
          {selectedDate.length >= 1 && <FormattedDate date={selectedDate[0]} />}
          {selectedDate.length >= 2 && <DateSeparator />}
          {selectedDate.length >= 2 && <FormattedDate date={selectedDate[1]} />}
        </Button>
      )}

      <Modal
        size="full"
        isOpen={isOpen}
        onClose={onClose}
        closeOnEsc
        closeOnOverlayClick
      >
        <ModalOverlay
          bg={'rgba(52, 64, 84, 0.7)'}
          backdropFilter={'blur(8px)'}
        />

        <ModalContent w={'0px'} h={'0px'}>
          <ModalBody background={'none'}>
            <Box
              pos={'absolute'}
              top={'50%'}
              left="50%"
              transform={'translate(-50%,-50%)'}
            >
              <DatePicker
                isOpen={true}
                onClose={onClose}
                type={type}
                userId={userId}
                timezone={timezone}
                dateEvents={dateEvents}
                onApplyDate={(d) => {
                  setSelectedDate(d);
                  onApplyDate?.(d);
                }}
              />
            </Box>
          </ModalBody>
        </ModalContent>
      </Modal>
    </Box>
  );
};

function DateSeparator() {
  return <Text as="span">&nbsp;&minus;&nbsp;</Text>;
}

function FormattedDate({ date }: { date: dateInfoProps['info'][number] }) {
  return (
    <Text as="span">
      {moment(`${date.year}-${date.month}-${date.day}`).format('ll')}
    </Text>
  );
}

export default DatePickerModal;
