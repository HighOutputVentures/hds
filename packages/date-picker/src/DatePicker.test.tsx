import { ThemeProvider } from "@highoutput/hds";
import { render, screen } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import { DatePicker } from "./DatePicker";
import DatePickerDropdown from "./DatePickerDropdown";
import DatePickerModal from "./DatePickerModal";

const MOCKDATA = [
  { date: new Date("2022-9-3"), title: "appointment", id: "1" },
  { date: new Date("2022-11-25"), title: "second", id: "3" },
  { date: new Date("2022-9-25"), title: "doctor", id: "2" },
  { date: new Date("2022-12-6"), title: "bd", id: "1" },
  { date: new Date("2022-12-23"), title: "bd", id: "1" },
];

describe("Datepicker pre-set range type Component", () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <DatePicker
          dateEvents={MOCKDATA}
          type={"pre-set ranges"}
          userId="1"
          onClose={jest.fn()}
          isOpen={true}
          timezone={"Asia/Manila"}
        />
      </ThemeProvider>,
    );
  });

  it("Should render", async () => {
    const RangeDatePicker = await screen.findAllByTestId(
      "hds.range-datepicker",
    );
    expect(RangeDatePicker).toHaveLength(1);
  });
});

describe("Datepicker single date type Component", () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <DatePicker
          dateEvents={MOCKDATA}
          type={"single date"}
          userId="1"
          onClose={jest.fn()}
          isOpen={true}
          timezone={"Asia/Manila"}
        />
      </ThemeProvider>,
    );
  });

  it("Should render", async () => {
    const SingleDatePicker = await screen.findAllByTestId(
      "hds.single-datepicker",
    );
    expect(SingleDatePicker).toHaveLength(1);
  });
});

describe("Datepicker dual date type Component", () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <DatePicker
          dateEvents={MOCKDATA}
          type={"dual dates"}
          userId="1"
          onClose={jest.fn()}
          isOpen={true}
          timezone={"Asia/Manila"}
        />
      </ThemeProvider>,
    );
  });

  it("Should render", async () => {
    const DualDatePicker = await screen.findAllByTestId("hds.dual-datepicker");
    expect(DualDatePicker).toHaveLength(1);
  });
});

describe("Datepicker modal Component", () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <DatePickerModal
          dateEvents={MOCKDATA}
          type={"dual dates"}
          userId="1"
          timezone={"Asia/Manila"}
        />
      </ThemeProvider>,
    );
  });

  it("Should render", async () => {
    const ModalDatePicker = await screen.findAllByTestId(
      "hds.modal-datepicker",
    );
    expect(ModalDatePicker).toHaveLength(1);
  });
});

describe("Datepicker dropdown Component", () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <DatePickerDropdown
          dateEvents={MOCKDATA}
          type={"dual dates"}
          userId="1"
          onClose={jest.fn()}
          isOpen={true}
          timezone={"Asia/Manila"}
        />
      </ThemeProvider>,
    );
  });

  it("Should render", async () => {
    const DropdownDatePicker = await screen.findAllByTestId(
      "hds.dropdown-datepicker",
    );
    expect(DropdownDatePicker).toHaveLength(1);
  });
});

describe("Snapshots", () => {
  it("(DatePicker Preset Range) Should match snapshot", () => {
    const component = renderer.create(
      <DatePicker
        dateEvents={MOCKDATA}
        type="pre-set ranges"
        userId="1"
        onClose={noop}
        isOpen
        timezone="Asia/Manila"
      />,
    );
    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
    component.unmount();
  });

  it("(DatePicker Single Date) Should match snapshot", () => {
    const component = renderer.create(
      <ThemeProvider>
        <DatePicker
          dateEvents={MOCKDATA}
          userId="1"
          type="single date"
          onClose={noop}
          timezone="Asia/Manila"
          isOpen
        />
      </ThemeProvider>,
    );
    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
    component.unmount();
  });

  it("(DatePicker Dual Date) Should match snapshot", () => {
    const component = renderer.create(
      <ThemeProvider>
        <DatePicker
          dateEvents={MOCKDATA}
          type="dual dates"
          userId="1"
          onClose={noop}
          isOpen
          timezone="Asia/Manila"
        />
      </ThemeProvider>,
    );
    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
    component.unmount();
  });

  it("(DatePickerModal) Should match snapshot", () => {
    const component = renderer.create(
      <ThemeProvider>
        <DatePickerModal
          dateEvents={MOCKDATA}
          type="dual dates"
          userId="1"
          timezone="Asia/Manila"
        />
      </ThemeProvider>,
    );
    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
    component.unmount();
  });

  it("(DatePickerDropdown) Should match snapshot", () => {
    const component = renderer.create(
      <ThemeProvider>
        <DatePickerDropdown
          dateEvents={MOCKDATA}
          timezone="Asia/Manila"
          type="dual dates"
          userId="1"
          onClose={noop}
          isOpen
        />
      </ThemeProvider>,
    );
    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
    component.unmount();
  });
});

function noop(..._args: any) {}
