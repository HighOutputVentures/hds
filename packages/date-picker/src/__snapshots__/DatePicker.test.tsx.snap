// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots (DatePicker Dual Date) Should match snapshot 1`] = `
Array [
  <div
    className="css-a3ggia"
    data-testid="hds.dual-datepicker"
  >
    <div
      className="chakra-stack css-n21gh5"
    >
      <div
        className="css-1xhj18k"
      >
        <div
          className="css-0"
        >
          <div
            className="chakra-stack css-4ihfsm"
          >
            <div
              className="chakra-stack css-1hfury2"
            >
              <svg
                className="chakra-icon chakra-icon css-1xogjj1"
                focusable={false}
                onClick={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  fill="currentColor"
                />
              </svg>
              <p
                className="chakra-text css-fzwid6"
              >
                March 2023
              </p>
              <svg
                className="chakra-icon chakra-icon css-1xogjj1"
                focusable={false}
                onClick={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              className="chakra-stack css-ni0b0p"
            >
              <p
                className="chakra-text css-54cw8p"
              >
                Select Date
              </p>
              <p
                className="chakra-text css-d6rt7l"
              >
                –
              </p>
              <p
                className="chakra-text css-1vwzryi"
              >
                Select Date
              </p>
            </div>
            <div
              className="css-1geqftd"
            >
              <div
                className="css-5f38dt"
              >
                Mo
              </div>
              <div
                className="css-5f38dt"
              >
                Tu
              </div>
              <div
                className="css-5f38dt"
              >
                We
              </div>
              <div
                className="css-5f38dt"
              >
                Th
              </div>
              <div
                className="css-5f38dt"
              >
                Fr
              </div>
              <div
                className="css-5f38dt"
              >
                Sat
              </div>
              <div
                className="css-5f38dt"
              >
                Su
              </div>
            </div>
            <div
              className="css-1c3l4rb"
            >
              <div
                className="css-1c6bxay"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1c958lr"
      >
        <div
          className="chakra-stack css-tth4vk"
        >
          <button
            className="chakra-button css-11a6hrq"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
          <button
            className="chakra-button css-1wl73lo"
            type="button"
          >
            Apply
          </button>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;

exports[`Snapshots (DatePicker Preset Range) Should match snapshot 1`] = `
<div
  className="css-4ivr56"
  data-testid="hds.range-datepicker"
>
  <div
    className="chakra-stack css-14stpeb"
  >
    <p
      className="chakra-text css-1c02hwd"
      onClick={[Function]}
    >
      Today
    </p>
    <p
      className="chakra-text css-1c02hwd"
      onClick={[Function]}
    >
      Yesterday
    </p>
    <p
      className="chakra-text css-1c02hwd"
      onClick={[Function]}
    >
      This week
    </p>
    <p
      className="chakra-text css-1c02hwd"
      onClick={[Function]}
    >
      Last week
    </p>
    <p
      className="chakra-text css-1c02hwd"
      onClick={[Function]}
    >
      This month
    </p>
    <p
      className="chakra-text css-1c02hwd"
      onClick={[Function]}
    >
      Last month
    </p>
    <p
      className="chakra-text css-1c02hwd"
      onClick={[Function]}
    >
      This year
    </p>
    <p
      className="chakra-text css-1c02hwd"
      onClick={[Function]}
    >
      Last year
    </p>
    <p
      className="chakra-text css-1c02hwd"
      onClick={[Function]}
    >
      All time
    </p>
  </div>
  <div
    className="chakra-stack css-n21gh5"
  >
    <div
      className="css-1xhj18k"
    >
      <div
        className="css-0"
      >
        <div
          className="chakra-stack css-4ihfsm"
        >
          <div
            className="chakra-stack css-rv21ye"
          >
            <svg
              className="chakra-icon chakra-icon css-1xogjj1"
              focusable={false}
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                fill="currentColor"
              />
            </svg>
            <p
              className="chakra-text css-fzwid6"
            >
              March 2023
            </p>
            <svg
              className="chakra-icon chakra-icon css-1xogjj1"
              focusable={false}
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            className="css-1geqftd"
          >
            <div
              className="css-5f38dt"
            >
              Mo
            </div>
            <div
              className="css-5f38dt"
            >
              Tu
            </div>
            <div
              className="css-5f38dt"
            >
              We
            </div>
            <div
              className="css-5f38dt"
            >
              Th
            </div>
            <div
              className="css-5f38dt"
            >
              Fr
            </div>
            <div
              className="css-5f38dt"
            >
              Sat
            </div>
            <div
              className="css-5f38dt"
            >
              Su
            </div>
          </div>
          <div
            className="css-1c3l4rb"
          >
            <div
              className="css-1c6bxay"
            />
          </div>
        </div>
      </div>
      <hr
        aria-orientation="vertical"
        className="chakra-divider css-12gj880"
      />
      <div
        className="css-0"
      >
        <div
          className="chakra-stack css-4ihfsm"
        >
          <div
            className="chakra-stack css-rv21ye"
          >
            <svg
              className="chakra-icon chakra-icon css-1xogjj1"
              focusable={false}
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                fill="currentColor"
              />
            </svg>
            <p
              className="chakra-text css-fzwid6"
            >
              March 2023
            </p>
            <svg
              className="chakra-icon chakra-icon css-1xogjj1"
              focusable={false}
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            className="css-1geqftd"
          >
            <div
              className="css-5f38dt"
            >
              Mo
            </div>
            <div
              className="css-5f38dt"
            >
              Tu
            </div>
            <div
              className="css-5f38dt"
            >
              We
            </div>
            <div
              className="css-5f38dt"
            >
              Th
            </div>
            <div
              className="css-5f38dt"
            >
              Fr
            </div>
            <div
              className="css-5f38dt"
            >
              Sat
            </div>
            <div
              className="css-5f38dt"
            >
              Su
            </div>
          </div>
          <div
            className="css-1c3l4rb"
          >
            <div
              className="css-1c6bxay"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-18xo0nj"
    >
      <div
        className="chakra-stack css-r99mt0"
      >
        <p
          className="chakra-text css-e5rdmw"
        >
          Select Date
        </p>
        <p
          className="chakra-text css-d6rt7l"
        >
          –
        </p>
        <p
          className="chakra-text css-9qj5qe"
        >
          Select Date
        </p>
      </div>
      <div
        className="chakra-stack css-1yfjwjq"
      >
        <button
          className="chakra-button css-1ynx0f"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="chakra-button css-5suas7"
          type="button"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshots (DatePicker Single Date) Should match snapshot 1`] = `
Array [
  <div
    className="css-a3ggia"
    data-testid="hds.single-datepicker"
  >
    <div
      className="chakra-stack css-n21gh5"
    >
      <div
        className="css-1xhj18k"
      >
        <div
          className="css-0"
        >
          <div
            className="chakra-stack css-4ihfsm"
          >
            <div
              className="chakra-stack css-1hfury2"
            >
              <svg
                className="chakra-icon chakra-icon css-1xogjj1"
                focusable={false}
                onClick={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  fill="currentColor"
                />
              </svg>
              <p
                className="chakra-text css-fzwid6"
              >
                March 2023
              </p>
              <svg
                className="chakra-icon chakra-icon css-1xogjj1"
                focusable={false}
                onClick={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              className="chakra-stack css-ni0b0p"
            >
              <p
                className="chakra-text css-1dw64w7"
              >
                Select Date
              </p>
              <div
                className="css-1tstedy"
              >
                Today
              </div>
            </div>
            <div
              className="css-1geqftd"
            >
              <div
                className="css-5f38dt"
              >
                Mo
              </div>
              <div
                className="css-5f38dt"
              >
                Tu
              </div>
              <div
                className="css-5f38dt"
              >
                We
              </div>
              <div
                className="css-5f38dt"
              >
                Th
              </div>
              <div
                className="css-5f38dt"
              >
                Fr
              </div>
              <div
                className="css-5f38dt"
              >
                Sat
              </div>
              <div
                className="css-5f38dt"
              >
                Su
              </div>
            </div>
            <div
              className="css-1c3l4rb"
            >
              <div
                className="css-1c6bxay"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1c958lr"
      >
        <div
          className="chakra-stack css-tth4vk"
        >
          <button
            className="chakra-button css-11a6hrq"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
          <button
            className="chakra-button css-1wl73lo"
            onClick={[Function]}
            type="button"
          >
            Apply
          </button>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;

exports[`Snapshots (DatePickerDropdown) Should match snapshot 1`] = `
Array [
  <button
    aria-controls="menu-list-:r8:"
    aria-expanded={false}
    aria-haspopup="menu"
    className="chakra-button chakra-menu__menu-button css-2f31vo"
    id="menu-button-:r8:"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    <span
      className="chakra-button__icon css-1wh2kri"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-onkibi"
        focusable={false}
        viewBox="0 0 14 14"
      >
        <path
          d="M10.8889,5.5 L3.11111,5.5 L3.11111,7.05556 L10.8889,7.05556 L10.8889,5.5 Z M12.4444,1.05556 L11.6667,1.05556 L11.6667,0 L10.1111,0 L10.1111,1.05556 L3.88889,1.05556 L3.88889,0 L2.33333,0 L2.33333,1.05556 L1.55556,1.05556 C0.692222,1.05556 0.00777777,1.75556 0.00777777,2.61111 L0,12.5 C0,13.3556 0.692222,14 1.55556,14 L12.4444,14 C13.3,14 14,13.3556 14,12.5 L14,2.61111 C14,1.75556 13.3,1.05556 12.4444,1.05556 Z M12.4444,12.5 L1.55556,12.5 L1.55556,3.94444 L12.4444,3.94444 L12.4444,12.5 Z M8.55556,8.61111 L3.11111,8.61111 L3.11111,10.1667 L8.55556,10.1667 L8.55556,8.61111 Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      className="css-xl71ch"
    >
      Select Date
    </span>
  </button>,
  <div
    className="css-r6z5ec"
    style={
      Object {
        "inset": "0 auto auto 0",
        "minWidth": "max-content",
        "position": "absolute",
        "visibility": "hidden",
      }
    }
  >
    <div
      aria-orientation="vertical"
      className="chakra-menu__menu-list css-b7mlum"
      data-testid="hds.dropdown-datepicker"
      id="menu-list-:r8:"
      onKeyDown={[Function]}
      role="menu"
      style={
        Object {
          "opacity": 0,
          "transform": "scale(0.8) translateZ(0)",
          "transformOrigin": "var(--popper-transform-origin)",
          "visibility": "hidden",
        }
      }
      tabIndex={-1}
    >
      <div
        className="css-a3ggia"
        data-testid="hds.dual-datepicker"
      >
        <div
          className="chakra-stack css-n21gh5"
        >
          <div
            className="css-1xhj18k"
          >
            <div
              className="css-0"
            >
              <div
                className="chakra-stack css-4ihfsm"
              >
                <div
                  className="chakra-stack css-1hfury2"
                >
                  <svg
                    className="chakra-icon chakra-icon css-1xogjj1"
                    focusable={false}
                    onClick={[Function]}
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                      fill="currentColor"
                    />
                  </svg>
                  <p
                    className="chakra-text css-fzwid6"
                  >
                    March 2023
                  </p>
                  <svg
                    className="chakra-icon chakra-icon css-1xogjj1"
                    focusable={false}
                    onClick={[Function]}
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  className="chakra-stack css-ni0b0p"
                >
                  <p
                    className="chakra-text css-54cw8p"
                  >
                    Select Date
                  </p>
                  <p
                    className="chakra-text css-d6rt7l"
                  >
                    –
                  </p>
                  <p
                    className="chakra-text css-1vwzryi"
                  >
                    Select Date
                  </p>
                </div>
                <div
                  className="css-1geqftd"
                >
                  <div
                    className="css-5f38dt"
                  >
                    Mo
                  </div>
                  <div
                    className="css-5f38dt"
                  >
                    Tu
                  </div>
                  <div
                    className="css-5f38dt"
                  >
                    We
                  </div>
                  <div
                    className="css-5f38dt"
                  >
                    Th
                  </div>
                  <div
                    className="css-5f38dt"
                  >
                    Fr
                  </div>
                  <div
                    className="css-5f38dt"
                  >
                    Sat
                  </div>
                  <div
                    className="css-5f38dt"
                  >
                    Su
                  </div>
                </div>
                <div
                  className="css-1c3l4rb"
                >
                  <div
                    className="css-1c6bxay"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="css-1c958lr"
          >
            <div
              className="chakra-stack css-tth4vk"
            >
              <button
                className="chakra-button css-11a6hrq"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="chakra-button css-1wl73lo"
                type="button"
              >
                Apply
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;

exports[`Snapshots (DatePickerModal) Should match snapshot 1`] = `
Array [
  <div
    className="css-0"
    data-testid="hds.modal-datepicker"
  >
    <button
      className="chakra-button css-2f31vo"
      onClick={[Function]}
      type="button"
    >
      <span
        className="chakra-button__icon css-1wh2kri"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-onkibi"
          focusable={false}
          viewBox="0 0 14 14"
        >
          <path
            d="M10.8889,5.5 L3.11111,5.5 L3.11111,7.05556 L10.8889,7.05556 L10.8889,5.5 Z M12.4444,1.05556 L11.6667,1.05556 L11.6667,0 L10.1111,0 L10.1111,1.05556 L3.88889,1.05556 L3.88889,0 L2.33333,0 L2.33333,1.05556 L1.55556,1.05556 C0.692222,1.05556 0.00777777,1.75556 0.00777777,2.61111 L0,12.5 C0,13.3556 0.692222,14 1.55556,14 L12.4444,14 C13.3,14 14,13.3556 14,12.5 L14,2.61111 C14,1.75556 13.3,1.05556 12.4444,1.05556 Z M12.4444,12.5 L1.55556,12.5 L1.55556,3.94444 L12.4444,3.94444 L12.4444,12.5 Z M8.55556,8.61111 L3.11111,8.61111 L3.11111,10.1667 L8.55556,10.1667 L8.55556,8.61111 Z"
            fill="currentColor"
          />
        </svg>
      </span>
      Select Date
    </button>
  </div>,
  <span />,
]
`;
